hutch_python/epics_arch.py

```diff
@@ -156,33 +156,93 @@ def create_arch_file(experiment, level=None, hutch=None, path=None, dry_run=Fals
         print_dry_run(experiment)
 
 
-def pull_cds_data(exp, run):
+def pull_cds_items(exp):
+    logger.debug("in client")
+
     """
-    Gather all user objects from the CDS tab in the questionnaire.
-    Parse objects and separate them based on type.
-    Display them in the console via PrettyTable.
+    Gather all user obejcts from the CDS tab in the questionnaire.
+    Parse objects and sperate them based on type.
+    Display them in the console vie PrettyTable.
 
     Parameters
     ----------
     exp: ``str``
         The experiment's name e.g. xppx1003221
     run: ''str''
         The run number e.g. run21
-
+    ----------
     Outputs
     -------
         PrettyTable visualization of cds objects
+    -------
+
     """
-    # pull run data from questionnaire api, then take the data and sort it
-    # create PrettyTable instance and if the values from the run data contain
-    # pcdssetup then put them into a seperate dictionary as they are cds items
-    logger.debug('pull_cds_items(%s)', exp)
-    client = QuestionnaireClient()
-    logger.debug("in cds items, run numb:", str(run[1]))
-    runDetails_Dict = client.getProposalDetailsForRun(str(run[0]), str(run[1]))
+    """
+    pull run data from questionnaire api, then take the data and sort it
+    create Pretty Table instance and if the values from the run data contain pcdssetup
+    then put them into a seperate dictionary as they are cds items
+    """
+
+    logger.debug('pull_cds_items:', exp)
+    try:
+        client = QuestionnaireClient()
+    except GSSError:
+        print('No kerberos token found; rerun after running kinit')
+        sys.exit()
+
+    formatted_run_id = ''
+    run_num = ''
+
+    # handle formatting of proposal id, want: e.g. X-10032
+    # Case -  Expected Format: e.g. xppx1003221
+    if re.match('^.{4}[0-9]{7}$', exp):
+        logger.debug("experiment name format")
+        run_num = 'run'+exp[-2:]
+        logger.debug('run num', run_num)
+        run_id = str(exp[3:-2])
```

Comment:
This works because you only need the bits from the start/end of a fixed-length string and you shouldn't change it, but you should be aware of capturing groups and how you can use these to handle arbitrarily complex regex matching/extraction. See https://docs.python.org/3/library/re.html

Reply:
No need to adjust this here! If it works, it works! (If it doesn't work, there's my suggestion though!)